R/compare_dfs.R

```diff
@@ -0,0 +1,70 @@
+get_first_col_classes <- function(dat) {
```

Comment:
Could this function simply be `vapply(dat class, character(1))`?

Reply:
It turns out that `vapply` will error when the function `FUN` returns a longer-than-expected result for an element of the input  `X`:

```
> tt <- data.frame(x = 1, y = 2)
> class(tt$y) <- c("a", "b")
> lapply(tt, class)
$x
[1] "numeric"

$y
[1] "a" "b"

> vapply(tt, class, character(1))
Error in vapply(tt, class, character(1)) : values must be length 1,
 but FUN(X[[2]]) result is length 2
```